src/controllers/v1/registration.js

```diff
@@ -186,13 +186,31 @@ const RegistrationController = {
    */
   delete: async (id, cleanObject) => {
     try {
+      // First find all attached returns.
+      const returns = await Return.findAll({where: {RegistrationId: id}});
+      // Create an array of the return ids.
+      const returnIds = returns.map((meatBaitReturn) => meatBaitReturn.id);
```

Comment:
This should be inside the transaction - as it is now, it's possible for a race-condition to cause a DB error. The user (or users) could issue the delete command twice (or more) at the same time and the results of the `findAll` will have been deleted by the time you get to to deleting them later.